src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -1487,6 +1486,8 @@ class Global(var currentSettings: Settings, var reporter: Reporter)
       compileUnitsInternal(units, fromPhase)
 
     private def compileUnitsInternal(units: List[CompilationUnit], fromPhase: Phase) {
+      def currentTime = java.util.concurrent.TimeUnit.NANOSECONDS.toMillis(System.nanoTime())
+
```

Comment:
Plus, why are we using `nanoTime`, but converting it back to milliseconds? Imho, the conversion should happen when displaying it, otherwise we have a mixture of nanoseconds and milliseconds in the same codebase.